craft_providers/lxd/lxd_instance.py

```diff
@@ -64,6 +78,55 @@ def __init__(
         else:
             self.lxc = lxc
 
+    def _set_instance_name(self) -> None:
```

Comment:
Probably this function should log (in debug) the final name, converted or not! because developers may want to "discover" the name of the instance for cases when there is need to handle them directly through LXC. What do you think?

Reply:
Yep, great idea! Added logging